pkg/controllers/managementuser/rbac/roletemplates/prtb_handler.go

```diff
@@ -265,19 +266,36 @@ func (p *prtbHandler) buildNamespaceBindings(prtb *v3.ProjectRoleTemplateBinding
 	}
 
 	// check if any of the aggregated CR grant updatepsa permission
+	psaCRName := fmt.Sprintf("%s-namespaces-psa", projectName)
 	psaRec := authorizer.AttributesRecord{
-		Verb:            "updatepsa",
+		Verb:            updatePSAVerb,
 		APIGroup:        management.GroupName,
 		Resource:        v3.ProjectResourceName,
 		Name:            prtb.ProjectName,
 		ResourceRequest: true,
 	}
 	if rbacAuth.RulesAllow(psaRec, cr.Rules...) {
-		namespacePSACR, err := rbac.BuildClusterRoleBindingFromRTB(prtb, namespacePSA)
+		// if rules allow user to use updatepsa,
+		// then we create the CR dedicated for the project.
+		psaCR := rbac.BuildClusterRole(psaCRName, prtb.RoleTemplateName, []rbacv1.PolicyRule{
+			{
+				APIGroups:     []string{management.GroupName},
+				Verbs:         []string{updatePSAVerb},
+				Resources:     []string{v3.ProjectResourceName},
+				ResourceNames: []string{projectName},
+			},
+		})
+		_, err := p.crClient.Create(psaCR)
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			return nil, err
+		}
+
+		// here we prepare the CRB to give updatepsa permission to the user.
```

Comment:
Doesn't seem like this comment doesn't add much context do the code that follows and can be removed
```suggestion
```